### PR TITLE
Starts work for merchant payments

### PR DESCRIPTION
**Pull request by dckesler**
### Description

Starts the merchant payments flow.

### Tested

Scanned a merchant payments QR code everything was correct. Can still scan other types of QR codes without errors.

### How others should test

This doesn't need QA yet.

### Backwards compatibility

Yes backwards compatible.

### CODE REVIEW

File: packages/mobile/src/firebase/firebase.ts [hunk 1]
```diff
@@ -284,6 +284,8 @@ export async function fetchRemoteFeatureFlags(): Promise<RemoteFeatureFlags | nu
       multiTokenShowHomeBalances: flags.multiTokenShowHomeBalances.asBoolean(),
       multiTokenUseSendFlow: flags.multiTokenUseSendFlow.asBoolean(),
       multiTokenUseUpdatedFeed: flags.multiTokenUseUpdatedFeed.asBoolean(),
+      paymentDeepLinkHandler: '',
```
Comment (gnardini) [inline packages/mobile/src/firebase/firebase.ts:287]: I can help you set up a feature flag so we can use the real code, reach out to me on Slack and we'll look into it :)

File: packages/mobile/src/app/reducers.ts [hunk 2]
```diff
@@ -83,6 +86,7 @@ const initialState = {
   multiTokenShowHomeBalances: FEATURE_FLAG_DEFAULTS.multiTokenShowHomeBalances,
   multiTokenUseSendFlow: FEATURE_FLAG_DEFAULTS.multiTokenUseSendFlow,
   multiTokenUseUpdatedFeed: FEATURE_FLAG_DEFAULTS.multiTokenUseUpdatedFeed,
+  paymentDeepLinkHandler: '' as PaymentDeepLinkHandler,
```
Comment (gnardini) [inline packages/mobile/src/app/reducers.ts:89]: The default should be read from `FEATURE_FLAG_DEFAULTS`

File: packages/mobile/src/qrcode/utils.ts [hunk 3]
```diff
@@ -158,3 +167,30 @@ export function* handleBarcode(
...
+  [key in PaymentDeepLinkHandler]: (barcode: QrCode) => Generator
+}
+
+const paymentDeepLinkHandlers: PaymentDeepLinkHandlers = {
+  '': paymentDeepLinkHandlerDefault,
+  merchant: paymentDeepLinkHandlerMerchant,
+}
+
+function* paymentDeepLinkHandlerDefault(barcode: QrCode) {
+  yield put(showError(ErrorMessages.QR_FAILED_INVALID_ADDRESS))
```
Comment (gnardini) [inline packages/mobile/src/qrcode/utils.ts:181]: I don't understand this, if the handler comes from a feature flag why would we put an invalid value there? It looks like the only real option is `merchant`, will there be other options in the future?

In any case, the empty string value we can rename to `disabled` maybe?

**Review by gnardini — COMMENTED** [review]
Just leaving a few early comments :)

File: packages/mobile/locales/en-US/translation.json [hunk 4]
```diff
@@ -1139,5 +1139,12 @@
...
-}
\ No newline at end of file
+  "celoDollarAmount_plural": "<0></0> Celo Dollars",
+  "merchant": "Merchant Test Screen",
+  "merchantMoneyEscrow": "You're about the send a direct payment to a merchant.",
+  "merchantPaymentNumberVerificationMessage": "You need to connect your number",
+  "merchantPaymentSubmitFail": "Your payment failed to submit",
+  "merchantPaymentSetup": "There was an error loading the merchant information, please try scanning the QRCode again",
+  "merchantPaymentSubmitSuccess": "Your payment successfully submitted",
+  "merchantWalletUnlockError": "TODO: soemthing went wrong when unlocking the wallet and/or DEK"
```
Comment (nicolasbrugneaux) [inline packages/mobile/locales/en-US/translation.json:1149]: note: still a todo

File: packages/mobile/src/firebase/remoteConfigValuesDefaults.e2e.ts [hunk 5]
```diff
@@ -39,4 +40,6 @@ export const REMOTE_CONFIG_VALUES_DEFAULTS: Omit<
   dappListApiUrl:
     'https://raw.githubusercontent.com/valora-inc/dapp-list/main/translations/valora-dapp-list-base.json',
   superchargeButtonType: SuperchargeButtonType.PillRewards,
+  // TODO: revert to .Disabled when ready to merge
+  paymentDeepLinkHandler: PaymentDeepLinkHandler.Merchant,
```
Comment (nicolasbrugneaux) [inline packages/mobile/src/firebase/remoteConfigValuesDefaults.e2e.ts:44]: note: don't forget this before merging

File: packages/mobile/src/firebase/remoteConfigValuesDefaults.ts [hunk 6]
```diff
@@ -43,4 +48,6 @@ export const REMOTE_CONFIG_VALUES_DEFAULTS: Omit<
   linkBankAccountEnabled: false,
   sentryTracesSampleRate: DEFAULT_SENTRY_TRACES_SAMPLE_RATE,
   superchargeButtonType: SuperchargeButtonType.PillRewards,
+  // TODO: revert to .Disabled when ready to merge
+  paymentDeepLinkHandler: PaymentDeepLinkHandler.Merchant,
```
Comment (nicolasbrugneaux) [inline packages/mobile/src/firebase/remoteConfigValuesDefaults.ts:52]: note: don't forget this before merging

File: packages/mobile/test/schemas.ts [hunk 7]
```diff
@@ -907,6 +908,18 @@ export const v29Schema = {
   },
 }
 
+export const v30Schema = {
```
Comment (nicolasbrugneaux) [inline packages/mobile/test/schemas.ts:911]: note for the reviewer: I wasn't sure I needed to add a new schema. Happy to remove if necessary

**Pull request closed without merging**